libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1901,6 +1896,20 @@ Expression const* SMTEncoder::leftmostBase(IndexAccess const& _indexAccess)
 	return base;
 }
 
+Expression const* SMTEncoder::innermostTuple(TupleExpression const& _tuple)
+{
+	solAssert(!_tuple.isInlineArray(), "");
+	TupleExpression const* tuple = &_tuple;
+	Expression const* expr = tuple;
+	while (tuple && tuple->components().size() == 1)
+	{
+		expr = tuple->components().front().get();
```

Comment:
What about `([x])`? You need the assertion in each loop iteration.

Reply:
Since the idea is to stop when the front component is not a Tuple anymore (not considering inline arrays), I think in that case it should just exit the while. Updated it in that way.